### DISCUSSION TITLE
WASM template project does not run properly when deployed

### DISCUSSION THREAD

**Discussion by Eldoir** [AvaloniaUI/Avalonia#14797] [Q&A]
## Describe the bug
Hello, maybe there is something I don't understand.
If I run the default project locally, everything's fine: get the url from the terminal that opens, it displays the splash screen, then the "Welcome to Avalonia" message a few seconds later.

However, if I run `dotnet publish` as [the doc says](https://docs.avaloniaui.net/docs/guides/platforms/how-to-use-web-assembly#deployment), then simply transfer the AppBundle folder that's been built to, say, https://mywebsite.com/AppBundle and access it, it fails to load the resources, so no css, no js, no logo.svg.
Pretty much the same as if I were to open the `AppBundle/index.html` on my computer without the terminal window acting as the "server".
What am I missing here? Do I need some sort of process running in the background on my server, pretty much like Node.js, in order to make it work?
Thanks and sorry if this is not a bug but just a noob question.

## To Reproduce
1. Create a new Web project using the Avalonia wizard
2. Publish it and post it online
3. Access it
4. See errors

## Expected behavior
I expect the online version to render the same as the version I launch in Debug or Release mode from VS.

## Screenshots
![image](https://github.com/AvaloniaUI/Avalonia/assets/30863193/0e5d9a7c-ae42-46af-a6bd-5bef867872a2)
![image](https://github.com/AvaloniaUI/Avalonia/assets/30863193/ab9bad40-77ac-4490-83ba-a1d690e837a4)

## Environment

- OS: Win10
- Avalonia-Version: 11.0.9
- Community MVVM Toolkit
- .NET 7.0

## Additional context
I thought that was related: https://github.com/AvaloniaUI/Avalonia/issues/14065 and tried the workaround because I'm targetting .NET 7.0 and not 8.0 (replace runMainAndExit by runMain), but to no avail.

**Comment by maxkatz6** [comment]
You can't just open index.html. You need to setup a static web server.
Locally, you can achieve that with `dotnet serve` - https://github.com/natemcmaster/dotnet-serve
With other hosting services, you need to contact them to get their static web server instructions.

**Comment by Eldoir** [comment]
Okay, so that was indeed not a defect 😅 Thanks for the quick reply and sorry again!
